libtenzir/builtins/functions/ip.cpp

```diff
@@ -0,0 +1,77 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/arrow_utils.hpp>
+#include <tenzir/concept/parseable/tenzir/ip.hpp>
+#include <tenzir/concept/parseable/tenzir/time.hpp>
+#include <tenzir/ip.hpp>
+#include <tenzir/series_builder.hpp>
+#include <tenzir/table_slice_builder.hpp>
+#include <tenzir/tql2/plugin.hpp>
+#include <tenzir/type.hpp>
+
+namespace tenzir::plugins::ip {
+
+namespace {
+
+class ip final : public function_plugin {
+public:
+  auto name() const -> std::string override {
+    return "tql2.ip";
+  }
+
+  auto make_function(invocation inv, session ctx) const
+    -> failure_or<function_ptr> override {
+    auto expr = ast::expression{};
+    TRY(argument_parser2::function("ip").add(expr, "<string>").parse(inv, ctx));
```

Comment:
Shouldn't this be this?

```suggestion
    TRY(argument_parser2::function("ip").add(expr, "<expr>").parse(inv, ctx));
```

Reply:
Would we have to apply this to all TQL2 functions then? The `int/uint` functions specify `<string|number>`, and  the `time` functions specifies `<string>`.